lib/internal/streams/writable.js

```diff
@@ -629,31 +636,30 @@ Writable.prototype.end = function(chunk, encoding, cb) {
   }
 
   if (err) {
-    // Do nothing...
-  } else if (!state.errored && !state.ending) {
+    process.nextTick(cb, err);
+  } else if (state.finished) {
+    process.nextTick(cb, new ERR_STREAM_ALREADY_FINISHED('end'));
+  } else if (state.destroyed) {
+    process.nextTick(cb, new ERR_STREAM_DESTROYED('end'));
+  } else if (state.ending) {
+    if (cb != null) {
+      state[kOnFinished] ??= [];
+      state[kOnFinished].push(cb);
```

Comment:
Can you re-run the benchmarks with keeping `kOnFinished` an array and not setting it to `null`? V8 can't optimize this if the type keeps changing.

Reply:
This should never occur in the benchmark.